kuma/dashboards/views.py

```diff
@@ -98,6 +94,13 @@ def revisions(request):
     if query_kwargs or exclude_kwargs:
         revisions = revisions.filter(**query_kwargs).exclude(**exclude_kwargs)
 
+    # prefetch_related needs to come after all filters have been applied to qs
+    revisions = revisions.prefetch_related('creator__bans',
```

Comment:
I think your intention is to get the `locale`, `slug` and `deleted` only from document.
I believe this should be something like

```python
revisions = (revisions.prefetch_related('creator__bans', 'akismet_submissions').prefetch_related(
                           Prefetch('document',
                                    queryset=Document.objects.only('deleted', 'locale', 'slug')
                           )
 ))

Reply:
Thanks @safwanrahman, I appreciate your taking a look at my untested WIP PR so quickly :)